### ISSUE TITLE
Performance optimization for interval.count.

### ISSUE THREAD

**Issue by jheer** [d3/d3-time#5]
[Datalib](https://github.com/vega/datalib) is now using d3-time for time calculations. When creating a histogram over large-ish tables with datetime fields, the `count` method can be called many times (100,000+) to compute bin indices. The multiple object allocations (`new Date()` calls) start to add up.

How about using temp Date objects instead? For example:
```
var a = new Date(),
    b = new Date();

if (count) interval.count = function(start, end) {
  a.setTime(+start), b.setTime(+end);
  floori(a), floori(b);
  return Math.floor(count(a, b));
};
```
(The temp dates might be moved out of the enclosing `newInterval` function scope.)

Some quick and dirty benchmarks in Chrome (Mac OS) suggest that this can reduce the running time by about 30%.

**Comment by mbostock** [comment]
A reasonable suggestion! Done.

**Comment by mbostock** [comment]
Released as 0.0.3.

**Comment by mbostock** [comment]
Unfortunately, this broke d3-time-format, which relies on overriding the `Date` class to a `UtcDate` implementation to use local time intervals and local time methods to perform operations in UTC. I may try to fix this in d3-time-format, but I’m leaving this open until this is resolved.

**Comment by mbostock** [comment]
Fixed in d3-time-format 0.0.2.
